### DISCUSSION TITLE
Why there is not any index for MapIndexes columns?

### DISCUSSION THREAD

**Discussion by jersiovic** [OrchardCMS/OrchardCore#15995] [Q&A]
We are making load tests in azure of our app bases on Orchard Core, and we experience timeouts for diffrent queries to Azure Sql Server. 
Trying to understand the reasons for the problem I've seen there is almost none db index defined on migrations for columns in MapIndex tables.

Is there any reason for that?

Thank you

**Comment by deanmarcussen** [comment]
Which `MapIndexes` are you not seeing indexes on?

I reviewed some of them a while ago (particularly `Autoroute` and `Alias`), and some others, but that does not mean there is not plenty of room for more improvements.

Especially useful if you are using profiler to examine the queries.

Can you show a couple of sample queries?

Also have you updated versions recently?

A new version of YesSql went in last week which included an updated MS Sql library which potentially may fix some timeout issues

**Comment by jersiovic** [comment]
Well we are using load tests with 100 sets of requests per second during 15 seconds.
This sets do it the following: 
1 -  Requesting a refresh token and access token with password flow 
2 - Request one document stored on db
3 - Request another document stored on db

Only a half of the refresh token request woks properly.

I've updated Orchard packages to yesterday packages at myget and now timeout dissapeared from logs, but it appears an error on opeind store looking user by name on bd

The initial requests works properly and at some point the server starts have more delay answering till it gives errors accessing user data. We tested it in Azure App Service with two instances runin with Linux docl¡ker and also locally in a development machine.

I found there is not any index  on OrchardCore.OpenId MapIndexes tables https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Migrations/OpenIdMigrations.cs

And the same for OrchardCore.Users https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Users/Migrations.cs

Then I grabed some table indexes on db finding no indexes.
But after reading your comment I 've searched by ".CreateIndex(" and found what you mention: There are more indexes created than what I though. But less than is expected. So I wil make a PRs adding those I would need.

No problem with that I will add those which I need asn will repeat tests. I was askings just to know how people are working without indexes. Maybe you are using autocreation of indexes in Azure or sth like that.

Thank you

**Comment by deanmarcussen** [comment]
> I've updated Orchard packages to yesterday packages at myget and now timeout dissapeared from logs

Cool, this is interesting. If the timeout reappears, please let us know.

Yes, make a pr for the indexes you need.

For infos, there were some issues around index key length (on both SQL and MySql) which is what I was mostly working on with `Autoroute` and `Alias`, and SQL Fields Indexing feature.

There's some tests over on a YesSql pr, https://github.com/sebastienros/yessql/pull/223, proving the maximum lengths (or you can just look at the `Alias` index to see what I mean)

So for anything being applied to an index we have to apply some max length logic to the value.

**Comment by Skrypt** [comment]
Even though I created the feature "SQL Fields Indexing" I strongly suggest that you make your owns as these table index can rapidly grow fast. It's always better to make a custom index per content type you create if that's what you are looking for.

Also, the "SQL Fields Indexing" misses badly an option to enable/disable it per field on a content type to make the feature more respectable.
